### PR TITLE
Testing

### LINKED ISSUE #58: Add a folder option for the `nmdb-graph-network` tool
**Is your enhancement request related to a problem? Please describe.**
It would be good if we can specify the icon folder for the `nmdb-graph-network`.  This would allow the "default" location to be documented as well as allow alternate icons to be located elsewhere instead of having to install all to that default location.

**Describe the solution you'd like**
The tool could have an option like `--icon-folder`.

**Describe alternatives you've considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by iburres**
Added option to declare a custom image path as a command-line argument.  

Resolves #58.

### CODE REVIEW

File: datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp [hunk 1]
```diff
@@ -181,36 +181,41 @@ class Tool : public nmdt::AbstractGraphTool
...
-          );
+          "icons",
+          NULL_SEMANTIC,
+          "Enable device icons in graph")
+        );
+      opts.addOptionalOption("icon-folder", std::make_tuple(
+          "icon-folder",
+          po::value<std::string>(),
+          "Set custom destination for alternate icons")
+        );
```
Comment (marshall-sg) [inline datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp:203]: This should be required and set the default value to be the current path which the tool uses.
```suggestion
      opts.addOptionalOption("icon-folder", std::make_tuple(
          "icon-folder",
          po::value<std::string>()->required()->default_value(nmfm.getConfPath()/"images"),
          "Set custom destination for alternate icons")
        );
```

My suggested edit requires you to declare `nmfm` which should be at the class level and something like `const nmcu::FileManager& nmfm {nmcu::FileManager::getInstance()};` (you can find/see this construct in other code in this repo).

File: datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp [hunk 2]
```diff
@@ -161,13 +161,13 @@ class IsRedundantEdge
...
 // Main code
 // =============================================================================
-
 class Tool : public nmdt::AbstractGraphTool
 {
   private:
     std::map<std::string, Vertex> vertexLookup;
     NetworkGraph graph;
     bool useIcons    {false};
+    bool useIconFolder {false};
```
Comment (marshall-sg) [inline datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp:170]: I don't understand what you are using this for.  The icon folder path is not optional.  Either the tool uses the default path or it uses an alternate path.

```suggestion
```

File: datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp [hunk 3]
```diff
@@ -358,51 +363,51 @@ class Tool : public nmdt::AbstractGraphTool
          "   hop_count"
          " FROM raw_ip_traceroutes");
 
+      useIcons = opts.exists("icons");
+      useIconFolder = opts.exists("icon-folder");
```
Comment (marshall-sg) [inline datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp:367]: Should not be needed.

```suggestion
```

File: datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp [hunk 4]
```diff
@@ -671,10 +675,15 @@ class Tool : public nmdt::AbstractGraphTool
...
-      label += "<td><font point-size=\"10\">" + name + "<br/>";
+      if (useIcons) {
+        label += getUseIconString(typeName);
+        label += "<td><font point-size=\"10\">" + name + "<br/>";
+      }
 
+      if (useIconFolder) {
+        label += getUseIconsCustomPath(typeName);
+        label += "<td><font point-size=\"10\">" + name + "<br/>";
+      }
```
Comment (marshall-sg) [inline datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp:686]: You should not need to make any edits to this code section.  The `getUseIconString()` function is what (currently) fills in the path for the icon folder, among other things.

File: datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp [hunk 5]
```diff
@@ -766,6 +775,44 @@ class Tool : public nmdt::AbstractGraphTool
...
+          }
+        }
+      }
+
+      label += iconPath + "\" scale=\"true\"/></TD>";
+
+      return label;
+    }
+
+
```
Comment (marshall-sg) [inline datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp:815]: This function is not needed and is a copy/paste of the `getUseIconString()` function, save the couple of `imagePath` related lines.

```suggestion
```

**Review by marshall-sg — CHANGES_REQUESTED** [review]
I have left several remarks regarding changes needed before I can accept this PR.

File: datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp (hunk 1 above)
Comment (iburres) [reply datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp:203]: Implemented suggested edits.

**Review by marshall-sg — APPROVED** [review]
Looks good.

**Pull request merged**